File: Frontend/src/app/servicio/reserva.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';
import { Observable } from 'rxjs';
import { Reserva } from '../modelos/reserva';

@Injectable({
  providedIn: 'root'
})
export class ReservaService {
  private baseUrl = environment.apiUrl;
  private url: string = `${this.baseUrl}/reservas`

  constructor(private http:HttpClient) { }

  listarReserva(): Observable<any>{
    return this.http.get<any>(this.url);
  }

  obtenerReserva(id_res:number):Observable<Reserva>{
    return this.http.get<Reserva>(`${this.url}/${id_res}`);
  }

  agregarReserva(reserva:Reserva):Observable<any>{
    return this.http.post<any>(this.url, reserva);
  }

  editarReserva(id_res:number, reserva:Reserva):Observable<any>{
    return this.http.put<any>(`${this.url}/${id_res}`,reserva);
  }

  eliminarReserva(id_res: number):Observable<any>{
    return this.http.delete<any>(`${this.url}/${id_res}`);
  }

  listarReservaEnable(): Observable<any> {
    return this.http.get<any>(`${this.url}/enable`);
  }

  eliminarReservaEnable(id_res: number): Observable<any> {
    return this.http.put<any>(`${this.url}/eliminar/${id_res}`, {});
  }

}

File: Backend/local/src/main/java/com/locales/service/ReservaService.java
package com.locales.service;

import java.util.Map;

import org.springframework.http.ResponseEntity;

import com.locales.model.Reserva;

public interface ReservaService {
	public ResponseEntity<Map<String, Object>> listarReserva();
	public ResponseEntity<Map<String, Object>> obtenerReservaPorId(Long id);
	public ResponseEntity<Map<String, Object>> agregarReserva(Reserva reserva);
	public ResponseEntity<Map<String, Object>> editarReserva(Reserva reserva, Long id);
	public ResponseEntity<Map<String, Object>> eliminarReserva(Long id);
	public ResponseEntity<Map<String, Object>> listarReservaEnable();
	public ResponseEntity<Map<String, Object>> eliminarReservaEnable(Long id);
	
}
